inc/providers/class-wp-sitemaps-taxonomies.php

```diff
@@ -62,27 +62,36 @@ public function get_url_list( $page_num, $taxonomy = '' ) {
 			return array();
 		}
 
+		/**
+		 * Filters the taxonomies URL list before it is generated.
+		 *
+		 * Passing a non-null value will effectively short-circuit the generation,
+		 * returning that value instead.
+		 *
+		 * @since 5.5.0
+		 *
+		 * @param array  $url_list The URL list. Default null.
+		 * @param string $taxonomy Taxonomy name.
+		 * @param int    $page_num Page of results.
+		 */
+		$url_list = apply_filters(
+			'wp_sitemaps_taxonomies_pre_url_list',
+			null,
+			$taxonomy,
+			$page_num
+		);
+
+		if ( null !== $url_list ) {
+			return $url_list;
+		}
+
 		$url_list = array();
 
 		// Offset by how many terms should be included in previous pages.
 		$offset = ( $page_num - 1 ) * wp_sitemaps_get_max_urls( $this->object_type );
 
-		$args = array(
-			'fields'                 => 'ids',
-			'taxonomy'               => $taxonomy,
-			'orderby'                => 'term_order',
-			'number'                 => wp_sitemaps_get_max_urls( $this->object_type ),
-			'offset'                 => $offset,
-			'hide_empty'             => true,
-
-			/*
-			 * Limits aren't included in queries when hierarchical is set to true (by default).
-			 *
-			 * @link: https://github.com/WordPress/WordPress/blob/5.3/wp-includes/class-wp-term-query.php#L558-L567
-			 */
-			'hierarchical'           => false,
-			'update_term_meta_cache' => false,
-		);
+		$args = $this->get_terms_query_args( $taxonomy );
```

Comment:
I would prefer `get_taxonomies_query_args` because it seems more consistent with the other functions